src/components/Section/ProposalVoting/ProposalVotingSection.tsx

```diff
@@ -68,56 +82,71 @@ const ProposalVotingSection = ({
         delegators,
         account,
         ownVotingPower,
-        delegatedVotingPower,
+        delegatedVp,
         voteDifference
       ),
-    [testData]
+    // TODO remove test data
+    [testData ? testData : [votes, choices, delegate, delegators, account, ownVotingPower, delegatedVp, voteDifference]]
+  )
+  const { votesByChoices, totalVotes } = useMemo(
+    () => getPartyVotes(delegators, votes, choices),
+    [delegators, votes, choices]
   )
-  const { votesByChoices, totalVotes } = useMemo(() => getPartyVotes(delegators, votes, choices), [testData])
 
+  loading = loading || accountState.loading || delegationState.loading || vpOnProposalState.loading
   return (
     <div className="DetailsSection__Content OnlyDesktop">
-      {testData && <VotingSectionTester testData={testData} />}
-      <Loader active={!loading && accountState.loading} />
-
-      {!account && (
-        <Button
-          basic
-          loading={accountState.loading}
-          disabled={accountState.loading}
-          onClick={() => accountState.select()}
-        >
-          {t('general.sign_in')}
-        </Button>
-      )}
+      {testData && <VotingSectionTester testData={testData} />
+        // TODO remove test data
+      }
+      {loading && <div className={'ProposalVotingSection__Loader'}>
+        <Loader active={loading} />
+      </div>}
 
-      {delegationsLabel && <DelegationsLabel {...delegationsLabel} />}
-
-      {(showChoiceButtons || changingVote) && (
-        <ChoiceButtons
-          choices={choices}
-          vote={vote}
-          votesByChoices={votesByChoices}
-          delegate={delegate}
-          delegateVote={delegateVote}
-          totalVotes={totalVotes}
-          onVote={onVote}
-          started={started}
-        />
-      )}
+      {!loading && (
+        <>
+          {!account && (
+            <Button
+              basic
+              loading={accountState.loading}
+              disabled={accountState.loading}
+              onClick={() => accountState.select()}
+            >
+              {t('general.sign_in')}
+            </Button>
+          )}
 
-      {votedChoice && !changingVote && <VotedChoiceButton {...votedChoice} />}
-
-      <VotingSectionFooter
-        vote={vote}
-        delegateVote={delegateVote}
-        hasDelegators={delegators && delegators.length > 0}
-        started={started}
-        finished={finished}
-        account={account}
-        changingVote={changingVote}
-        onChangeVote={onChangeVote}
-      />
+          {delegationsLabel && <DelegationsLabel {...delegationsLabel} />}
+
+          {(showChoiceButtons || changingVote) && (
```

Comment:
I know you didn't change this but... maybe the check of `changingVote` can be in `showChoiceButtons`?

Reply:
The thing is `showChoiceButtons` comes from the getConfig fn, and if we want it to include `changingVote` we'd need to pass it as a parameter. It'd be better if we can avoid calling it each time `changingVote` changes.

Maybe `showChoiceButtons` should be called `showVotingOptions` ? 🤔

Reply:
Oh ok, maybe you can keep this name.